dali/operators/image/remap/cvcuda/warp_perspective.cc

```diff
@@ -0,0 +1,241 @@
+  case the matrix can be placed on the GPU.)doc",
+                           std::vector<float>({}), true, true)
+    .AddOptionalArg("border_mode",
+                    "Border mode to be used when accessing elements outside input image.\n"
+                    "Supported values are: \"constant\", \"replicate\", "
+                    "\"reflect\", \"reflect_101\", \"wrap\".",
+                    "constant")
+    .AddOptionalArg("interp_type", "Type of interpolation used.", DALI_INTERP_LINEAR)
+    .AddOptionalArg("pixel_origin", R"doc(Pixel origin. Possible values: "corner", "center".
+
+Defines which part of the pixel (upper-left corner or center) is interpreted as its origin.
+This value impacts the interpolation result. To match OpenCV, please pick "center".))doc", "corner")
+    .AddOptionalArg<float>("fill_value",
+                           "Value used to fill areas that are outside the source image when the "
+                           "\"constant\" border_mode is chosen.",
+                           std::vector<float>({}))
+    .AddOptionalArg<bool>("inverse_map",
+                          "If set to true (default), the matrix is interpreted as "
+                          "destination to source coordinates mapping. "
+                          "Otherwise it's interpreted as source to destination "
+                          "coordinates mapping.", true);
+
+
+class WarpPerspective : public nvcvop::NVCVSequenceOperator<StatelessOperator> {
+ public:
+  explicit WarpPerspective(const OpSpec &spec)
+      : nvcvop::NVCVSequenceOperator<StatelessOperator>(spec),
+        border_mode_(nvcvop::GetBorderMode(spec.GetArgument<std::string>("border_mode"))),
+        interp_type_(nvcvop::GetInterpolationType(spec.GetArgument<DALIInterpType>("interp_type"))),
+        inverse_map_(spec.GetArgument<bool>("inverse_map")),
+        fill_value_arg_(spec.GetArgument<std::vector<float>>("fill_value")),
+        ocv_pixel_(OCVCompatArg(spec.GetArgument<std::string>("pixel_origin"))) {
+    matrix_data_.SetContiguity(BatchContiguity::Contiguous);
+  }
+
+  bool ShouldExpandChannels(int input_idx) const override {
+    return true;
+  }
+
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+  float4 GetFillValue(int channels) const {
+    if (fill_value_arg_.size() > 1) {
+      if (channels > 0) {
+        if (channels == static_cast<int>(fill_value_arg_.size())) {
+          float4 fill_value{0, 0, 0, 0};
+          memcpy(&fill_value, fill_value_arg_.data(), sizeof(decltype(fill_value)));
+          return fill_value;
+        } else {
+          DALI_FAIL(make_string(
+              "Number of values provided as a fill_value should match the number of channels.\n"
+              "Number of channels: ",
+              channels, ". Number of values provided: ", fill_value_arg_.size(), "."));
+        }
+      } else {
+        DALI_FAIL("Only scalar fill_value can be provided when processing data in planar layout.");
```

Comment:
Is that CV-CUDA limitation or ours?

Reply:
Lets say CV-CUDA. The problem is CV-CUDA doesn't support planar layouts at all (in this op), and it only accepts single fill_value for the whole batch, so cannot provide different fill value for each plane

Reply:
We could launch the planes separately, but whatever.